lib/exabgp/bgp/message/update/attribute/mprnlri.py

```diff
@@ -69,29 +69,19 @@ def packed_attributes (self, negotiated):
 					# .packed and not .pack()
 					nexthop = nlri.nexthop.ton(negotiated,nlri.afi)
 
-			# mpunli[afi,safi][nexthop] = nlri
-			mpnlri.setdefault((nlri.afi.pack(),nlri.safi.pack()),{}).setdefault(nexthop,[]).append(nlri.pack(negotiated))
-
-		for (pafi,psafi),data in mpnlri.iteritems():
-			for nexthop,nlris in data.iteritems():
-				payload = \
-					pafi + psafi + \
-					chr(len(nexthop)) + nexthop + \
-					chr(0) + ''.join(nlris)
+			# mpunli[nexthop] = nlri
+			mpnlri.setdefault(nexthop,[]).append(nlri.pack(negotiated))
 
-				if self._len(payload) <= maximum:
+		for nexthop,nlris in mpnlri.iteritems():
+			payload = ''.join([self.afi.pack(), self.safi.pack(), chr(len(nexthop)), nexthop, chr(0)])
+			for nlri in nlris:
+				if self._len(payload + nlri) > maximum:
 					yield self._attribute(payload)
+					payload = ''.join([self.afi.pack(), self.safi.pack(), chr(len(nexthop)), nexthop, chr(0), nlri])
 					continue
+				payload  = ''.join([payload, nlri])
 
-				# This will not generate an optimum update size..
-				# we should feedback the maximum on each iteration
-
-				for nlri in nlris:
-					yield self._attribute(
-						pafi + psafi +
-						chr(len(nexthop)) + nexthop +
-						chr(0) + nlri
-					)
+			yield self._attribute(payload)
 
 	def pack (self, negotiated):
 		return ''.join(self.packed_attributes(negotiated))
```

Comment:
Easy to fix. I do have a question though do you know if the pack function on mprlnri is called anywhere? I couldn't actually find anyplace in the code that was using it myself. The problem with this pack function is that if it is ever called and used anywhere it can still generate multiple nlris. If its not used I think the best option would be to remove it.